### DISCUSSION TITLE
How can i use group by in a many-to-many relationship to sum values

### DISCUSSION THREAD

**Discussion by bardzh** [prisma/prisma#6517] [Q&A]
im using **Prisma 2.20.1**
i need to get the sum of values in certain categories that are linked from two different tables in a many to many relation,
something like this
```prisma
model User {
  id      String @id
}

model Category{
  id       String @id
  name String
}

model UseCategory{
  userId String     @relation(fields: [userId], references: [id])
  categoryId String @relation(fields: [categoryId], references: [id])
  value Int
}
```
this is what i need
```
[{
 category: { Category }
 sum: {value: Int }
}]
```
if i run groupBy function
```ts
prisma.userCategory.groupBy({
          where:{
            userId
          },
          by: ['categoryId'],
          sum: {
            value: true
          }
   })
```
this returns: 
```
[
  {
    categoryId: '578206f3-5a1b-4e62-b531-77791de6a7c9',
    sum: { value: 0 }
  },
]
```

i need to somehow create `include` in the group by so i can get category information with it.

**Comment by ryands17** [comment]
@bardiaheshmati 👋 

In this case, you would need to fetch the Categories and then map each category to the value here as Prisma doesn't directly support including the relation.

**Reply by bardzh** [reply]
that's what i ended up doing, my dataset is not big so i'm not worried much but if the list gets bigger then it becomes very expensive.

**Reply by tdadadavid** [reply]
my dataset is big, and doing this approach is expensive, other ideas anyone

**Reply by losefor** [reply]
### Use Raw SQL instead of Prisma in large datasets since it will be way more efficient 
Prisma is not for big datasets it might be easier to work with but unfortunately Prisma under the hood makes a lot of calls to the DB which makes it not suitable for large projects if you have a large dataset simply **write everything in raw SQL**

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Comment by leonardomgt** [comment]
This is definitely a must-have feature! 
Do you have any prospects for adding the feature?
